### DISCUSSION TITLE
Optional Scaling/Stretching to aspect ratio of Moonlight client

### DISCUSSION THREAD

**Discussion by roundaboutluke** [LizardByte/community#151] [Sunshine]
Related Discord thread - https://discord.com/channels/804382334370578482/1066008272672407675

Essentially, GFE will scale/stretch the image of a 16:9 or 16:10 host resolution aspect ratio to match that of the Moonlight client resolution in the event that the aspect ratio is different so that it fills the screen.

Although host resolution can be set to match the client resolution and thus aspect ratio, this causes a lot of games to then letterbox back to 16:9/16:10 aspect ratio, so the stretching for weird screened clients (like the MacBook Pro with notch, or the iPhone 14) is beneficial in these cases to make the most of the screen real estate.

I believe this feature should be optional as understand that scaling/stretching isn't always desired.

The similar discussion to automatically match client resolution is not what I am looking for here, hence starting this thread!

**Comment by Albertfanmingo** [comment]
I wrote this as it's own post, but it very much fits, and is another example of how this feature would be helpful:

In current Sunshine, if Moonlight's requested resolution does not fit the aspect ratio of the host, The video is letterboxed.


![Screenshot_20250420_194515_Moonlight](https://github.com/user-attachments/assets/45587431-0b51-4daa-a72f-273de437a482)
_If I stream from a 32:9 monitor to Moonlight asking for a 16:9 resolution, black bars are added to the top and bottom._


This is a fine way of doing things, assuming that every screen we're streaming to has square pixels. The New Nintendo 3DS does not. It has double the pixel density horizontally than vertically, with a resolution of 800x240 (10:3) but an aspect ratio of 5:3. This means that if we attempt to stream at it's native resolution, Sunshine incorrectly letterboxes the video, ruining it's output.


![Screenshot from 2025-04-20 20-08-0723](https://github.com/user-attachments/assets/59858b69-b335-40c7-9719-ecd258ffe1f4)
_The output when Moonlight requests an 800x240 resolution from a 800x480 monitor._

![Screenshot from 2025-04-20 20-08-fsda0723](https://github.com/user-attachments/assets/c2e8c191-f5c3-4cc7-81c3-0ea079c5073b)
_The output on N3DS._


Having an option to instead stretch the video to fit the output resolution fixes this.


![Screenshot from 2025-04-20 20-08-072](https://github.com/user-attachments/assets/a6abfabc-113f-4baf-b5fe-9b9640084794)
_An output where the video is not letterboxed but instead stretched to fit the requested resolution._

![Screenshot from 2025-04-20 20-08-07](https://github.com/user-attachments/assets/831ee3b6-845d-466b-8bfa-363c67a937fa)
_How it would look on N3DS._

**Comment by Barborica-Alexandru** [comment]
I only just noticed this discussion and I essentially made a duplicate. Maybe they can be merged together? https://github.com/orgs/LizardByte/discussions/627

I'm still fine with using GFE as long as scaling is not available in moonlight, I just wish the streaming component could be separated from GFE so I don't have to keep the whole thing installed.
